src/marqo/s2_inference/hf_utils.py

```diff
@@ -16,15 +28,105 @@ def __init__(self, *args, **kwargs) -> None:
+        self.model_auth = kwargs.get(InferenceParams.model_auth, None)
 
     def load(self) -> None:
 
-        self.model = AutoModelForSentenceEmbedding(self.model_name).to(self.device)
-        self.tokenizer = AutoTokenizer.from_pretrained(self.model_name)
+        model_location_presence = ModelProperties.model_location in self.model_properties
+        path = self.model_properties.get("localpath", None) or self.model_properties.get("url", None)
+        # HF models can be loaded from 3 entries: path (url or localpath), model_name, or model_location
+        if (path is not None) + (self.model_name is not None) + (model_location_presence is True) != 1:
+            raise InvalidModelPropertiesError("Exactly one of (`localpath`/`url`) or `model_location`, `name` can be specified"
+                                              " in `model_properties` for `hf` models as they conflict with each other in model loading."
+                                              " Please ensure that exactly one of these is specified in `model_properties` and retry.")
+        elif path is not None:
+            if validators.url(path) is True:
+                self.model_path = download_model(url = path, download_dir=ModelCache.hf_cache_path)
+            elif os.path.isdir(path) or os.path.isfile(path):
+                self.model_path = path
+        elif self.model_name is not None:
+            # Loading from structured huggingface repo directly, token is required directly
+            self.model_path = self.model_name
+        elif model_location_presence is True:
+            # This is a special case for huggingface models, where we can load a model directory from a repo
+            if ("hf" in self.model_properties["model_location"]) and ("repo_id" in self.model_properties["model_location"]["hf"]) and \
+                ("filename" not in self.model_properties["model_location"]["hf"]):
+                return self._load_from_private_hf_repo()
+            else:
+                self.model_path = self._download_from_repo()
+
+        # We need to do extraction here if necessary
+        self.model_path = validate_huggingface_archive(self.model_path)
+
+        self.model = AutoModelForSentenceEmbedding(self.model_path).to(self.device)
+        try:
+            self.tokenizer = AutoTokenizer.from_pretrained(self.model_path)
+        except (OSError, ValueError, RuntimeError) as e:
+            raise InvalidModelPropertiesError(
+                f"Marqo encounters error loading the Hugging Face model = `{self.model_path}` using AutoTokenizer "
+                f"Please ensure that the model is a valid Hugging Face model and retry.\n"
+                f" Original error message = {e}")
+        except (HTTPError, ConnectionError) as e:
+            raise ModelDownloadError(
+                f"Marqo encounters ConnectionError loading the Hugging Face model = `{self.model_path}` using AutoTokenizer. "
```

Comment:
Can you please use past tense for these errors. For example, "Marqo encountered..." rather than "Marqo encounters..."